backend_new/src/services/user.service.ts

```diff
@@ -782,6 +783,73 @@ export class UserService {
     return rawUser;
   }
 
+  /*
+    gets and formats user data to be handed to the csv builder helper
+    this data will be emailed to the requesting user
+  */
+  async export(requestingUser: User): Promise<SuccessDTO> {
+    const users = await this.list(
+      {
+        page: 1,
+        limit: 'all',
+        filter: [
+          {
+            isPortalUser: true,
+          },
+        ],
+      },
+      requestingUser,
+    );
+
+    const parsedUsers = users.items.reduce((accum, user) => {
+      const roles: string[] = [];
+      if (user.userRoles?.isAdmin) {
+        roles.push('Administrator');
+      }
+      if (user.userRoles?.isPartner) {
+        roles.push('Partner');
+      }
+      if (user.userRoles?.isJurisdictionalAdmin) {
+        roles.push('Jurisdictional Admin');
+      }
+
+      const listingNames: string[] = [];
+      const listingIds: string[] = [];
+
+      user.listings?.forEach((listing) => {
+        listingNames.push(listing.name);
+        listingIds.push(listing.id);
+      });
+
+      accum[user.id] = {
+        'First Name': user.firstName,
+        'Last Name': user.lastName,
+        Email: user.email,
+        Role: roles.join(', '),
+        'Date Created': dayjs(user.createdAt).format('MM-DD-YYYY HH:mmZ[Z]'),
+        Status: user.confirmedAt ? 'Confirmed' : 'Unconfirmed',
+        'Listing Names': listingNames.join(', '),
+        'Listing Ids': listingIds.join(', '),
+        'Last Logged In': dayjs(user.lastLoginAt).format(
+          'MM-DD-YYYY HH:mmZ[Z]',
+        ),
+      };
+      return accum;
+    }, {});
+
+    const csvData = buildFromIdIndex(parsedUsers);
+    await this.emailService.sendCSV(
+      requestingUser.jurisdictions,
+      requestingUser,
+      csvData,
+      'User Export',
+      'a user export',
```

Comment:
This is a bit clunky of wording for the user export. I wonder if it should be "The attached file is an export of all users."

Reply:
nice call out, updated!